### PR TITLE
Core/Packet Update structure for SMSG_ITEM_TIME_UPDATE

### PR DESCRIPTION
**Pull request by ProjectcoreDevs**
Update opcode CMSG_OPEN_ITEM.

### CODE REVIEW

**Comment by funjoker** [comment]
looks good, but update it to new opcode structure and change DEFINE_SERVER_OPCODE_HANDLER

**Comment by et65** [comment]
This is updated to the new opcode structure...

File: src/server/game/Server/Packets/ItemPackets.h [hunk 1]
```diff
@@ -70,6 +70,17 @@ namespace WorldPackets
             uint32 Amount = 0;
         };
 
+        class ItemTimeUpdate final : public ServerPacket
+        {
+        public:
+            ItemTimeUpdate() : ServerPacket(SMSG_ITEM_TIME_UPDATE, 8 + 4) { }
+
+            WorldPacket const* Write() override;
+
```
Comment (Vincent-Michael) [inline src/server/game/Server/Packets/ItemPackets.h]: pls rename in:


public class ClientItemTimeUpdate
{
    public ulong ItemGuid    { get; set; }
    public uint DurationLeft { get; set; }
}

File: src/server/game/Entities/Item/Item.cpp [hunk 2]
```diff
@@ -1048,10 +1049,10 @@ void Item::SendTimeUpdate(Player* owner)
     if (!duration)
         return;
 
-    WorldPacket data(SMSG_ITEM_TIME_UPDATE, (8+4));
-    data << GetGUID();
-    data << uint32(duration);
-    owner->GetSession()->SendPacket(&data);
+    WorldPackets::Item::ItemTimeUpdate itemTimeUpdate;
+    itemTimeUpdate.GUID = GetGUID();
+    itemTimeUpdate.Duration = uint32(duration);
```
Comment (Intel) [inline src/server/game/Entities/Item/Item.cpp]: Typecasting is not required if you're not dealing with bytebuffer

File: src/server/game/Entities/Item/Item.cpp [hunk 3]
```diff
@@ -17,17 +17,18 @@
...
+#include "Opcodes.h"
+#include "Player.h"
 #include "SpellMgr.h"
 #include "SpellInfo.h"
 #include "ScriptMgr.h"
-#include "ConditionMgr.h"
-#include "Player.h"
-#include "Opcodes.h"
+#include "WorldPacket.h"
 #include "WorldSession.h"
```
Comment (jackpoz) [inline src/server/game/Entities/Item/Item.cpp]: what are these header include changes for ?

Comment (joschiwald) [inline src/server/game/Entities/Item/Item.cpp]: legibility

**Pull request closed without merging**